### PR TITLE
Flight task auto params

### PR DESCRIPTION
**Pull request by Stifael**
@bkueng can you have a look?

### CODE REVIEW

File: src/modules/mc_pos_control/PositionControl.hpp [hunk 1]
```diff
@@ -182,42 +188,23 @@ class PositionControl
...
-	param_t MPC_Z_VEL_MAX_DN_h { PARAM_INVALID };
-	param_t MPC_Z_VEL_MAX_UP_h { PARAM_INVALID };
-	param_t MPC_THR_HOVER_h { PARAM_INVALID };
-	param_t MPC_THR_MAX_h { PARAM_INVALID };
-	param_t MPC_THR_MIN_h { PARAM_INVALID };
-	param_t MPC_MANTHR_MIN_h { PARAM_INVALID };
-	param_t MPC_TILTMAX_AIR_h { PARAM_INVALID };
-	param_t MPC_MAN_TILT_MAX_h { PARAM_INVALID };
+	DEFINE_PARAMETERS(
+	        (ParamFloat<px4::params::MPC_THR_MAX>) MPC_THR_MAX,
```
Comment (bkueng) [inline src/modules/mc_pos_control/PositionControl.hpp]: I strongly suggest you stick to the usual naming pattern for members: something like `_maximum_thrust`. Otherwise we have another inconsistency in the code-base.
However it's not part of this PR.

File: src/modules/mc_pos_control/PositionControl.hpp [hunk 2]
```diff
@@ -77,12 +77,20 @@ struct Constraints {
...
+	PositionControl(ModuleParams *parent);
+	~PositionControl() = default;
+
+	/**
+	 *	Overwrites certain parameters.
+	 *	Overwrites are required for unit-conversion.
+	 *	This method should only be called if parameters
+	 *	have been updated.
+	 */
+	void overwriteParams();
```
Comment (bkueng) [inline src/modules/mc_pos_control/PositionControl.hpp]: Can you make `ModuleParams::updateParams` virtual, and override it here instead?

File: src/modules/mc_pos_control/PositionControl.cpp [hunk 3]
```diff
@@ -285,58 +263,23 @@ void PositionControl::_velocityController(const float &dt)
...
-	param_get(MPC_THR_MAX_h, &MPC_THR_MAX);
-	param_get(MPC_THR_MIN_h, &MPC_THR_MIN);
-	param_get(MPC_MANTHR_MIN_h, &MPC_MANTHR_MIN);
-	param_get(MPC_TILTMAX_AIR_h, &MPC_TILTMAX_AIR);
-	MPC_TILTMAX_AIR = math::radians(MPC_TILTMAX_AIR);
-	param_get(MPC_MAN_TILT_MAX_h, &MPC_MAN_TILT_MAX);
-	MPC_MAN_TILT_MAX = math::radians(MPC_MAN_TILT_MAX);
-
+	// Tilt needs to be in radians
+	MPC_TILTMAX_AIR.set(math::radians(MPC_TILTMAX_AIR.get()));
```
Comment (bkueng) [inline src/modules/mc_pos_control/PositionControl.cpp]: This is prone to errors: if for some reason `overwriteParams` is not called, or several times, or you use `MPC_TILTMAX_AIR.commit()`, you will have completely wrong values. I suggest using a separate class member instead (I used the `_deg` suffix to make clear which one is in degrees). Not super nice, but much less error prone.

File: src/modules/mc_pos_control/PositionControl.hpp (hunk 1 above)
Comment (Stifael) [reply src/modules/mc_pos_control/PositionControl.hpp]: I discuss that with @MaEtUgR, and we came to the conclusion that it is more confusing to redefine the parameter names. For instance, the parameter in one module can have a complete different name in another module. In addition, from the name it is also not clear who is the owner of that parameters. 

Reusing the name of the metadata removes the consistent redefinition of parameter names...

File: src/modules/mc_pos_control/PositionControl.hpp (hunk 2 above)
Comment (Stifael) [reply src/modules/mc_pos_control/PositionControl.hpp]: sure, that could work as well

File: src/modules/mc_pos_control/PositionControl.hpp (hunk 1 above)
Comment (bkueng) [reply src/modules/mc_pos_control/PositionControl.hpp]: That needs a bigger upstream discussion then, you cannot simply make that decision in an unrelated PR. And if decided someone needs to go through the codebase and update all other places.
In addition, there are also reasons against this, for example:
- allows to use more descriptive variable names, since params are restricted to 16 chars.
- naming consistency with other other class members: after all, a parameter is just a class member like any other
- naming conflicts/inconsistencies with macros/globals: typically macros or globals use all uppercase. This is also confusing for newcomers

Comment (Stifael) [reply src/modules/mc_pos_control/PositionControl.hpp]: > That needs a bigger upstream discussion then, you cannot simply make that decision in an unrelated PR

I don't expect that this [PR\(https://github.com/PX4/Firmware/pull/9253) (= base of this PR) gets merged soon, but you are right, probably these param name changes should not be added to this PR. In the meantime I will initiate a discussion about parameter naming. 

> after all, a parameter is just a class member like any other

Not really. There should be a distinction between class members that change regularly (like setpoints) and class members that are parameters which do not change frequently. 

> naming conflicts/inconsistencies with macros/globals: typically macros or globals use all uppercase. This is also confusing for newcomers

I am sure that can be easily solved by adding a px4 convention page in the px4 devguide. I think the fact that there is no formal px4 convention page is the root cause of the confusion  because a beginner has no chance to know what is legacy cold vs expected px4 implementation.

Comment (hamishwillee) [reply src/modules/mc_pos_control/PositionControl.hpp]: > I am sure that can be easily solved by adding a px4 convention page in the px4 devguide. I think the fact that there is no formal px4 convention page is the root cause of the confusion because a beginner has no chance to know what is legacy code vs expected px4 implementation.

That sounds like a good idea. Probably as a sub-page of [Contribution](https://dev.px4.io/en/contribute/#contribution). All we have at the moment is that you should use astyle for formatting.

Can you create an [issue](https://github.com/PX4/Devguide/issues) outlining the main conventions you are interested in - in bullet point form and ideally with examples. We can then drag in others and iterate.
